tdrs-backend/tdpservice/parsers/schema_defs/ssp/m1.py

```diff
@@ -14,92 +14,244 @@
+                    ]),
               Field(item="2", name='COUNTY_FIPS_CODE', type='string', startIndex=19, endIndex=22,
-                    required=True, validators=[]),
+                    required=True, validators=[
+                        validators.isNumber(),
+                    ]),
               Field(item="4", name='STRATUM', type='string', startIndex=22, endIndex=24,
-                    required=True, validators=[]),
+                    required=True, validators=[
+                        validators.isInStringRange(0, 99),
+                    ]),
               Field(item="6", name='ZIP_CODE', type='string', startIndex=24, endIndex=29,
-                    required=True, validators=[]),
+                    required=True, validators=[
+                        validators.isNumber(),
+                    ]),
               Field(item="7", name='DISPOSITION', type='number', startIndex=29, endIndex=30,
-                    required=True, validators=[]),
+                    required=True, validators=[
+                        validators.oneOf([1, 2]),
+                    ]),
               Field(item="8", name='NBR_FAMILY_MEMBERS', type='number', startIndex=30, endIndex=32,
-                    required=True, validators=[]),
+                    required=True, validators=[
+                        validators.isInLimits(1, 2),  # ?? doc says 1-2, but t1 has isNumber
+                    ]),
               Field(item="9", name='FAMILY_TYPE', type='number', startIndex=32, endIndex=33,
-                    required=True, validators=[]),
+                    required=True, validators=[
+                        validators.isInLimits(1, 3),
+                    ]),
               Field(item="10", name='TANF_ASST_IN_6MONTHS', type='number', startIndex=33, endIndex=34,
-                    required=True, validators=[]),
+                    required=True, validators=[
+                        validators.isInLimits(1, 3),
+                    ]),
               Field(item="11", name='RECEIVES_SUB_HOUSING', type='number', startIndex=34, endIndex=35,
-                    required=True, validators=[]),
+                    required=True, validators=[
+                        validators.isInLimits(1, 2),  # diff than t1 (1-3)
+                    ]),
               Field(item="12", name='RECEIVES_MED_ASSISTANCE', type='number', startIndex=35, endIndex=36,
-                    required=True, validators=[]),
+                    required=True, validators=[
+                        validators.isInLimits(1, 2),
+                    ]),
               Field(item="13", name='RECEIVES_FOOD_STAMPS', type='number', startIndex=36, endIndex=37,
-                    required=True, validators=[]),
+                    required=True, validators=[
+                        validators.isInLimits(1, 2),  # diff than t1 (0-2)
+                    ]),
               Field(item="14", name='AMT_FOOD_STAMP_ASSISTANCE', type='number', startIndex=37, endIndex=41,
```

Comment:
great question, glad you brought it up! i'd definitely like @ADPennington to weigh in on this - we discussed this a bit and my takeaway was that for some fields it may be useful to preserve the leading zeros, but for many fields it's more useful to have an int value in the database when querying. this kind of goes against an earlier decision that we'd like to keep the resulting data structure strictly consistent with the legacy system. for me it was easiest to not make any changes to the underlying model, and if we'd like to change anything we can (hopefully) have a smaller migration

Reply:
@jtimpe @elipe17 yeah, think we lean towards ints for amounts and such and special fields like stratum we try to store as str to preserve leading zeros. there may be inconsistencies here and there but it may be hard to catch until parsing is complete. we can revisit. but so far during testing im seeing consistency.